src/features/tags/store.ts

```diff
@@ -48,6 +48,24 @@ const tagsSlice = createSlice({
       state.tagList.isLoading = false;
       state.tagList.items.push(remoteItem(tag.id, { data: tag }));
     },
+    tagDeleted: (state, action: PayloadAction<number>) => {
+      const tagId = action.payload;
+      const tagListItem = state.tagList.items.find((item) => item.id === tagId);
+
+      if (tagListItem) {
+        tagListItem.deleted = true;
+      }
+
+      for (const personId in state.tagsByPersonId) {
+        const item = state.tagsByPersonId[personId].items.find((item) => {
+          return item.id === tagId;
+        });
+
+        if (item) {
+          item.deleted = true;
+        }
```

Comment:
Let me start by saying that I don't think there's anything necessarily wrong with this solution, but I think it may be using `deleted` unnecessarily. The purpose of the `deleted` property on items is to prevent loading of data, and that should not be a problem here. So the way I would have approached this is with a simpler(?) solution:

```suggestion
        state.tagsByPersonId[personId].items = state.tagsByPersonId[personId].items.filter(
          (item) => item.id != tagId
        );
```

This just _removes_ the tag from the person, similar to what would happen if we untagged the person (which is essentially what happens when we delete the tag).

Reply:
I remember a small bird whispering to my ear "But what would we use the deleted property for on a tag? I'm not sure this makes sense?" while I was writing this haha. I will change it and now I know how to handle it in the future :)